x/dex/keeper/msg_server.go

```diff
@@ -59,14 +59,20 @@ func (k msgServer) Deposit(goCtx context.Context, msg *types.MsgDeposit) (*types
 func (k msgServer) Withdrawl(goCtx context.Context, msg *types.MsgWithdrawl) (*types.MsgWithdrawlResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	token0, token1, createrAddr, receiverAddr, err := k.WithdrawlVerification(goCtx, *msg)
+	// validate msg
+	if err := msg.ValidateBasic(); err != nil {
```

Comment:
@teddyknox requested that these validations happen in the `ValidateBasic()` method on the message objects. The `sdk.Msg` interface that these all conform to requires that `ValidateBasic()` only returns an error, so it can't also return the addresses.

Reply:
If you and Teddy are onboard that this makes sense I'm satisfied.